### PR TITLE
make secretbox methods return nullable

### PR DESCRIPTION
**Pull request by wiyarmir**
this matches the actual api contract of the underlying tweetnacl java implementation

### CODE REVIEW

File: tweetnacl-multiplatform/src/jvmMain/kotlin/net/avianlabs/solana/tweetnacl/TweetNaCl.kt [hunk 1]
```diff
@@ -20,9 +20,9 @@ internal actual fun generateKeyInternal(seed: ByteArray): Ed25519Keypair {
 
 internal actual fun secretBoxInternal(secretKey: ByteArray): TweetNaCl.SecretBox =
   object : TweetNaCl.SecretBox {
-    override fun box(message: ByteArray, nonce: ByteArray): ByteArray =
+    override fun box(message: ByteArray, nonce: ByteArray): ByteArray? =
       TweetNaclFast.SecretBox(secretKey).box(message, nonce)
 
-    override fun open(box: ByteArray, nonce: ByteArray): ByteArray =
+    override fun open(box: ByteArray, nonce: ByteArray): ByteArray? =
       TweetNaclFast.SecretBox(secretKey).open(box, nonce)
```
Comment (jszmltr) [inline tweetnacl-multiplatform/src/jvmMain/kotlin/net/avianlabs/solana/tweetnacl/TweetNaCl.kt:27]: Doesn't this throw instead of returning null ?

**Review by jszmltr — APPROVED** [review]

File: tweetnacl-multiplatform/src/jvmMain/kotlin/net/avianlabs/solana/tweetnacl/TweetNaCl.kt (hunk 1 above)
Comment (jszmltr) [reply tweetnacl-multiplatform/src/jvmMain/kotlin/net/avianlabs/solana/tweetnacl/TweetNaCl.kt:27]: Maybe we can runCatching getOrNull here ?

Comment (jszmltr) [reply tweetnacl-multiplatform/src/jvmMain/kotlin/net/avianlabs/solana/tweetnacl/TweetNaCl.kt:27]: Ideally with some logger 👀 :¯\_(ツ)_/¯

Comment (wiyarmir) [reply tweetnacl-multiplatform/src/jvmMain/kotlin/net/avianlabs/solana/tweetnacl/TweetNaCl.kt:27]: nope

Comment (jszmltr) [reply tweetnacl-multiplatform/src/jvmMain/kotlin/net/avianlabs/solana/tweetnacl/TweetNaCl.kt:27]: Ah ok. It was throwing nullpointer exception with the result being null 👍

Comment (wiyarmir) [reply tweetnacl-multiplatform/src/jvmMain/kotlin/net/avianlabs/solana/tweetnacl/TweetNaCl.kt:27]: sorry the nope was at the throw comment 😂

Comment (wiyarmir) [reply tweetnacl-multiplatform/src/jvmMain/kotlin/net/avianlabs/solana/tweetnacl/TweetNaCl.kt:27]: this is a library, I think the functionality should be minimal. I'm doing all those things in the app

**Pull request merged**